test/functional/healthcheck.test.js

```diff
@@ -55,13 +60,52 @@ test('proxy requests originating from behind the broker client', t => {
         });
       });
 
+      t.test('server client-health with connected client', t => {
+        request({url: serverClientHealth, json: true }, (err, res) => {
+          if (err) { return t.threw(err); }
+
+          t.equal(res.statusCode, 200, '200 statusCode');
+          t.equal(res.body['ok'], true, '{ ok: true } in body');
+          t.end();
+        });
+      });
+
+
+      t.test('server client-health after client disconnected', t => {
+        client.close();
+        setTimeout(() => {
+          request({url: serverClientHealth, json: true }, (err, res) => {
+            if (err) { return t.threw(err); }
+
+            t.equal(res.statusCode, 200, '200 statusCode');
+            t.equal(res.body['ok'], false, '{ ok: false } in body');
+            t.end();
+          });
+        }, 100);
```

Comment:
Perhaps we could use the `close` event instead of waiting with setTimeout?

```javascript
client.io.on('close', () => {
  request({url: serverClientHealth, json: true }, (err, res) => {
    if (err) { return t.threw(err); }
    t.equal(res.statusCode, 200, '200 statusCode');
    t.equal(res.body['ok'], false, '{ ok: false } in body');
    t.end();
  });
});
```

If the socket never closes then we'd get slower failure, but success would be faster at least.

(i'm not sure this would actually work, just an idea)